packages/core/src/shared/utilities/workspaceUtils.ts

```diff
@@ -25,14 +25,17 @@ type GitIgnoreRelativeAcceptor = {
     acceptor: GitIgnoreAcceptor
 }
 
+const alwaysIncludeFiles = ['gradle/wrapper/gradle-wrapper.jar']
 export class GitIgnoreFilter {
     private acceptors: GitIgnoreRelativeAcceptor[]
+    private alwaysInclude: string[]
 
-    private constructor(acceptors: GitIgnoreRelativeAcceptor[]) {
+    private constructor(acceptors: GitIgnoreRelativeAcceptor[], alwaysInclude: string[]) {
         this.acceptors = acceptors
+        this.alwaysInclude = alwaysInclude
     }
 
-    public static async build(gitIgnoreFiles: vscode.Uri[]): Promise<GitIgnoreFilter> {
+    public static async build(gitIgnoreFiles: vscode.Uri[], alwaysInclude: string[] = []): Promise<GitIgnoreFilter> {
```

Comment:
so now we have a way to override the gitignore, which itself is an override for the "collect files" logic. and none of this is documented?